feathr_project/feathr/_preprocessing_pyudf_manager.py

```diff
@@ -0,0 +1,151 @@
+import inspect
+import os
+from pathlib import Path
+from typing import List, Optional, Union
+import pickle
+from jinja2 import Template
+from feathr.source import HdfsSource
+
+# Some metadata that are only needed by Feathr
+FEATHR_PYSPARK_METADATA = 'generated_feathr_pyspark_metadata'
+# UDFs that are provided by users and persisted by Feathr into this file.
+# It will be uploaded into the Pyspark cluster as a dependency for execution
+FEATHR_CLIENT_UDF_FILE_NAME = 'client_udf_repo.py'
+# Pyspark driver code that is executed by the Pyspark driver
+FEATHR_PYSPARK_DRIVER_FILE_NAME = 'feathr_pyspark_driver.py'
+FEATHR_PYSPARK_DRIVER_TEMPLATE_FILE_NAME = 'feathr_pyspark_driver_template.py'
+# Feathr provided imports for pyspark UDFs all go here
+PROVIDED_IMPORTS = ['\nfrom pyspark.sql import SparkSession, DataFrame\n'] + \
+                   ['from pyspark.sql.functions import *\n']
+
+
+class _PreprocessingPyudfManager(object):
+    """This class manages Pyspark UDF preprocessing related artifacts, like UDFs from users, the pyspark_client etc.
```

Comment:
Briefly explain how the manager works will be good.

Reply:
added to the methods